### ISSUE TITLE
[Due for payment 2025-02-18] [$250] Workspace- Device backspace and system back button have different function for Category/Tag

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#54378]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.77-3
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):** testpayment935+817@gmail.com
**Issue reported by:** Applause - Internal Team
**Component:** Workspace Settings

## Action Performed:
1. Navigate to https://staging.new.expensify.com/
2. Log in to an any account
3. Navigate to WS (Create one if not any)
4. Navigate to Category /Tag list then select some Categories or Tags
5. Click system back button at the top left and notice the selection is dismissed
6. Select some categories/ tags again
7. Click the device back button and notice it navigate back to WS settings

## Expected Result:
Device backspace and system back button have the same function

## Actual Result:
Device backspace and system back button have different function for Category/Tag

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [x] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details> 

https://github.com/user-attachments/assets/62351b3e-e25c-41b6-aefd-d558e8ab1ac1

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021873799168918490516</li>
        <li>Upwork Job ID: 1873799168918490516</li>
        <li>Last Price Increase: 2025-01-27</li>
<li>Automatic offers: </li>
<ul>
<li>layacat | Contributor | 105922187</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @alexpensify</details>

**Comment by FitseTLT** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Workspace- Device backspace and system back button have different function for Category/Tag

### What is the root cause of that problem?
This is because on pressing back button we only dismiss the selection mode instead of navigating back here
https://github.com/Expensify/App/blob/582c6f14e2ae53fc4d12d355d37fa007d9796925/src/pages/workspace/categories/WorkspaceCategoriesPage.tsx#L363-L369
### What changes do you think we should make in order to solve the problem?

If we want to be consistent we can go back in all cases so change it to 
```
onBackButtonPress={() => {
                        Navigation.goBack(backTo);
                    }}
```
Do the same for tag settings page too
### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?


### What alternative solutions did you explore? (Optional)

**Comment by alexpensify** [comment]
There is no update yet; still on my list.

**Comment by alexpensify** [comment]
Still on my testing list

**Comment by alexpensify** [comment]
@hungvu193 when you get a chance, can you review if this proposal will fix the issue? Thanks! 

--- 

## Heads up, I will be offline until Friday, January 3, 2025, and will not actively watch over this GitHub during that period.

If this GitHub requires an urgent update, please ask for help in the #expensify-open-source Slack Room. If the inquiry can wait, I'll review it when I return online.

**Comment by VektorTech** [comment]
# Proposal

In [App/src/pages/workspace/categories/WorkspaceCategoriesPage.tsx](https://github.com/Expensify/App/blob/main/src/pages/workspace/categories/WorkspaceCategoriesPage.tsx)

```js
// on selection mode is enabled
window.history.pushState(null, null, window.location.href);
.
.
.
useEffect(() => {
  const terminateSelection = () => {
    if (!selectionMode?.isEnabled) return Navigation.goBack(backTo);
    window.history.go(1);
    setSelectedCategories({});
    turnOffMobileSelectionMode();
  }
  window.addEventListener("popstate", terminateSelection);
  return () => { window.removeEventListener("popstate", terminateSelection); }
}, [selectionMode?.isEnabled, backTo]);
```

**Comment by hungvu193** [comment]
Hi @VektorTech, please follow [our proposal template here](https://github.com/Expensify/App/blob/main/contributingGuides/PROPOSAL_TEMPLATE.md)

**Comment by hungvu193** [comment]
@FitseTLT I believe we don't wanna remove that feature

> If we want to be consistent we can go back in all cases so change it to

**Comment by hungvu193** [comment]
Thanks @layacat, I don't think `hardwareBackPress` supports mWeb.

**Comment by hungvu193** [comment]
> On mWeb, this is default web behavior, we can't prevent the popstate because it was handle by the browser.

Yep. I agree with you on this point.

@alexpensify This issue happened with both Android and mWeb. On Android, it looks like we can't handle pressing back. (The only way that is available on mWeb, I think, is to prompt the user that they're leaving.) 

Otherwise we can just be going back anyway on back press, like @FitseTLT 's proposal above. What do you think?

**Comment by hungvu193** [comment]
little bump @alexpensify

**Comment by alexpensify** [comment]
Thanks for flagging! Tomorrow, let me start a discussion to confirm with the team in the Open Source Slack room.

**Comment by hungvu193** [comment]
Not overdue

**Comment by alexpensify** [comment]
Note for myself and make sure this one doesn't fall off the task list.

**Comment by hungvu193** [comment]
Not overdue

**Comment by hungvu193** [comment]
Not overdue

**Comment by hungvu193** [comment]
Little bump @alexpensify

**Comment by hungvu193** [comment]
Thank you 🚀

**Comment by hungvu193** [comment]
Not overdue. I think let's fix Android native. Wdyt?

**Comment by alexpensify** [comment]
Yeah, let's move forward with that plan! Will one of these proposals fix the issue on Android only? Thanks for checking!

**Comment by hungvu193** [comment]
Yes. Let's go with @layacat 's [proposal](https://github.com/Expensify/App/issues/54378#issuecomment-2567228518)

🎀👀🎀 C+ reviewed

**Comment by grgia** [comment]
Confirming with android users the expected behavior - https://expensify.slack.com/archives/C01GTK53T8Q/p1738148452178419?thread_ts=1737687275.614879&cid=C01GTK53T8Q

**Comment by hungvu193** [comment]
> hey guys, do we need to address lint changed below?
> 
> Namespace imports from @libs are not allowed. Use named imports instead. Example: import { method } from "@libs/module"                no-restricted-syntax
> There are so many of them and it doesn't relate to my current PR logic

Unfortunately, we can't merge the PR without addressing all the lints, we need to align with the rules, please address all the lints. Ty 🙇

**Comment by alexpensify** [comment]
@hungvu193 can you please review if this warning is accurate or an error? Thanks!

**Comment by alexpensify** [comment]
Contributor: @layacat paid $250 via Upwork
Contributor+: @hungvu193 owed $250 via NewDot

**Comment by garrettmknight** [comment]
$250 approved for @hungvu193
